pkg/cloudprovider/types.go

```diff
@@ -57,18 +58,22 @@ type CloudProvider interface {
 	// Availability of types or zone may vary by provisioner or over time.  Regardless of
 	// availability, the GetInstanceTypes method should always return all instance types,
 	// even those with no offerings available.
-	GetInstanceTypes(context.Context, *v1alpha5.Provisioner) ([]*InstanceType, error)
+	GetInstanceTypes(context.Context, *v1alpha5.Provisioner) (InstanceTypes, error)
 	// Name returns the CloudProvider implementation name.
 	Name() string
 }
 
 type NodeRequest struct {
 	Template            *scheduling.NodeTemplate
-	InstanceTypeOptions []*InstanceType
+	InstanceTypeOptions InstanceTypes
 }
 
+// +k8s:deepcopy-gen=true
+type InstanceTypes []*InstanceType
```

Comment:
I went back and forth on whether to type this at all but I can see use cases for having methods on this type so I kept it

Reply:
i.e. `Requirements()` similar to `Offerings`